Package release workflow is not working

<!-- This form is for other issue types specific to the WooCommerce plugin. This is not a support portal. -->

The [Package Release workflow](https://github.com/woocommerce/woocommerce/actions/workflows/package-release.yml) is not working - I believe this is to do with `npm` requiring 2fa. See thread: p1697554592761099-slack-C03CPM3UXDJ

Logging the issue here as it doesn't look like it has been saved as an issue yet.

The error message indicates an unclean working tree. I'm wondering if a pnpm lock file change has been created as a result of the Setup WooCommerce Monorepo step.

```
ERR_PNPM_GIT_UNCLEAN  Unclean working tree. Commit or stash changes first.
```

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -20,18 +20,15 @@ jobs:
         permissions:
           contents: read
         steps:
-            - name: Test
-              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run == 'true' && '--dry-run' ) || '' }} --skip-install
+            - uses: actions/checkout@v3
             
-            # - uses: actions/checkout@v3
-            
-            # - name: Setup WooCommerce Monorepo
-            #   uses: ./.github/actions/setup-woocommerce-monorepo
+            - name: Setup WooCommerce Monorepo
+              uses: ./.github/actions/setup-woocommerce-monorepo
 
-            # - name: Clean working directory
-            #   run: git checkout pnpm-lock.yaml # in case for whatever reason the lockfile is out of sync, there won't be interference with npm publish.
+            - name: Clean working directory
+              run: git checkout pnpm-lock.yaml # in case for whatever reason the lockfile is out of sync, there won't be interference with npm publish.
 
-            # - name: Execute script
-            #   run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} --skip-install
-            #   env:
-            #       NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            - name: Execute script
+              run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run == 'true' && '--dry-run' ) || '' }} --skip-install
+              env:
+                  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -6,7 +6,7 @@ on:
                 description: 'Enter a specific package to release, or releases separated by commas, ie @woocommerce/components,@woocommerce/number. Leaving this input blank will release all eligible packages.'
                 required: false
                 default: '-a'
-            dry_run:
+            tags:
                 description: 'Perform all checks but don\'t actually make the release.'
                 type: boolean
                 required: false

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -6,7 +6,7 @@ on:
                 description: 'Enter a specific package to release, or releases separated by commas, ie @woocommerce/components,@woocommerce/number. Leaving this input blank will release all eligible packages.'
                 required: false
                 default: '-a'
-            tags:
+            dry_run:
                 description: 'Perform all checks but don\'t actually make the release.'
                 type: boolean
                 required: false

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -7,10 +7,9 @@ on:
                 required: false
                 default: '-a'
             dry_run:
-                description: 'Perform all checks but don\'t actually make the release.'
+                description: 'Perform all checks but dont actually make the release.'
                 type: boolean
                 required: false
-                default: false
 
 permissions: {}
 

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -7,7 +7,7 @@ on:
                 required: false
                 default: '-a'
             dry_run:
-                description: 'Perform all checks but dont actually make the release.'
+                description: 'Perform a dry run. Complete all steps but dont actually make the release.'
                 type: boolean
                 required: false
 
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: echo ${{ github.event.inputs.dry_run }}
+              run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.packages && '--dry-run' || '' }} --skip-install ${{ github.event.inputs.packages && '--dry-run' || '' }}
             
             # - uses: actions/checkout@v3
             

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -20,15 +20,18 @@ jobs:
         permissions:
           contents: read
         steps:
-            - uses: actions/checkout@v3
+            - name: Test
+              run: echo ${{ github.event.inputs.dry_run }}
             
-            - name: Setup WooCommerce Monorepo
-              uses: ./.github/actions/setup-woocommerce-monorepo
+            # - uses: actions/checkout@v3
+            
+            # - name: Setup WooCommerce Monorepo
+            #   uses: ./.github/actions/setup-woocommerce-monorepo
 
-            - name: Clean working directory
-              run: git checkout pnpm-lock.yaml # in case for whatever reason the lockfile is out of sync, there won't be interference with npm publish.
+            # - name: Clean working directory
+            #   run: git checkout pnpm-lock.yaml # in case for whatever reason the lockfile is out of sync, there won't be interference with npm publish.
 
-            - name: Execute script
-              run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} --skip-install
-              env:
-                  NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+            # - name: Execute script
+            #   run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} --skip-install
+            #   env:
+            #       NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,6 +21,9 @@ jobs:
             - name: Setup WooCommerce Monorepo
               uses: ./.github/actions/setup-woocommerce-monorepo
 
+            - name: Clean working directory
+              run: git checkout pnpm-lock.yaml # in case for whatever reason the lockfile is out of sync, there won't be interference with npm publish.
+
             - name: Execute script
               run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} --skip-install
               env:

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run === 'true' && '--dry-run' ) || '' }} --skip-install
+              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run == 'true' && '--dry-run' ) || '' }} --skip-install
             
             # - uses: actions/checkout@v3
             

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -6,6 +6,11 @@ on:
                 description: 'Enter a specific package to release, or releases separated by commas, ie @woocommerce/components,@woocommerce/number. Leaving this input blank will release all eligible packages.'
                 required: false
                 default: '-a'
+            dry_run:
+                description: 'Perform all checks but don\'t actually make the release.'
+                type: boolean
+                required: false
+                default: false
 
 permissions: {}
 

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.dry_run && '--dry-run' || '' }} --skip-install
+              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run && '--dry-run' ) || '' }} --skip-install
             
             # - uses: actions/checkout@v3
             

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run && '--dry-run' ) || '' }} --skip-install
+              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ ( github.event.inputs.dry_run === 'true' && '--dry-run' ) || '' }} --skip-install
             
             # - uses: actions/checkout@v3
             

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.packages && '--dry-run' || '' }} --skip-install ${{ github.event.inputs.packages && '--dry-run' || '' }}
+              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.packages && '--dry-run' || '' }} --skip-install
             
             # - uses: actions/checkout@v3
             

--- a/.github/workflows/package-release.yml
+++ b/.github/workflows/package-release.yml
@@ -21,7 +21,7 @@ jobs:
           contents: read
         steps:
             - name: Test
-              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.packages && '--dry-run' || '' }} --skip-install
+              run: echo ./tools/package-release/bin/dev publish ${{ github.event.inputs.packages }} --branch=${{ github.ref_name }} ${{ github.event.inputs.dry_run && '--dry-run' || '' }} --skip-install
             
             # - uses: actions/checkout@v3
             

